cl/alerts/tasks.py

```diff
@@ -168,25 +200,58 @@ def send_docket_alert(
         delete_redis_semaphore("ALERTS", make_alert_key(d_pk))
         return
 
-    # Notify every user that's subscribed to this alert.
-    messages = make_alert_messages(
-        d, new_des, email_addresses, recap_email_recipients
+    docket_alert_messages = make_alert_messages(
+        d,
+        new_des,
+        docket_alert_recipients,
+        first_email=False,
+        auto_subscribe=False,
+    )
+    re_first_time_auto_subscribe_messages = make_alert_messages(
+        d,
+        new_des,
+        recap_email_auto_subscribe_recipients,
+        first_email=True,
+        auto_subscribe=True,
+    )
+    re_first_time_no_auto_subscribe_messages = make_alert_messages(
+        d,
+        new_des,
+        recap_email_no_auto_recipients,
+        first_email=True,
+        auto_subscribe=False,
+    )
+
+    messages = (
+        docket_alert_messages
+        + re_first_time_auto_subscribe_messages
+        + re_first_time_no_auto_subscribe_messages
     )
     connection = get_connection()
     connection.send_messages(messages)
 
     # Work completed. Tally, log, and clean up
-    tally_stat("alerts.docket.alerts.sent", inc=len(email_addresses))
+    tally_stat("alerts.docket.alerts.sent", inc=len(messages))
     DocketAlert.objects.filter(docket=d).update(date_last_hit=now())
+
+    # Webhooks recipients are handled separately, send it to current docket
+    # alerts recipients + first time recap.email recipients with auto_subscribe
+    # set to true.
+    webhook_recipients = (
+        docket_alert_recipients + recap_email_auto_subscribe_recipients
+    )
+    # Send the docket to webhook
+    webhook_email_recipients = [email[0] for email in webhook_recipients]
+    send_docket_to_webhook.delay(d_pk, since, webhook_email_recipients)
```

Comment:
Passing the email addresses down to the webhook is an interesting optimization, but I think it's not worth the confusion ("Why are we sending email addresses to the webhook?") or the tethering ("Why do we need to send alerts before we can send webhooks?").

Also, the name of the function is a bit odd. Shouldn't it be `send_docket_alert_webhooks`?

So, with those two changes, I think the signature here should change to:
```
def send_docket_alert_webhooks(d_pk: int, since: datetime):
```

Reply:
Changed, now it's not necessary to pass the recipients to `send_docket_alert_webhooks`.

About the question:
`Why do we need to send alerts before we can send webhooks?`

We don't need to send alerts before we send webhooks but we need to get the recipients of docket alerts before sending the webhooks because when getting the recipients we also create docket alerts for recap email users with auto-subscribe turned on, so we also include those docket alert subscriptions (first-time notifications for a case) when sending webhooks.